### ISSUE TITLE
Multicluster fails on EKS cluster when created with helm chart

### ISSUE THREAD

**Issue by yfried** [linkerd/linkerd2#5359]
### What is the issue?
`linkerd check` reports:
```
× all gateway mirrors are healthy
        liveness checks failed for crosscluster-test-2
    see https://linkerd.io/checks/#l5d-multicluster-gateways-endpoints for hints
```

### How can it be reproduced?
Deployed using `linkerd2` (`v2.9.0`) and `linkerd2-multicluster` (`edge-20.12.1`)
The link was deployed using cli according to tutorial

### Logs, error output, etc

```
❯ klf --since 1m linkerd-service-mirror-crosscluster-test-2-57f7795fcd-x9rp4 -c service-mirror
time="2020-12-09T13:02:03Z" level=warning msg="Gateway returned unexpected status 502. Marking as unhealthy" probe-key=crosscluster-test-2
time="2020-12-09T13:02:57Z" level=info msg="Received: RepairEndpoints" apiAddress="https://XXXXX.us-east-2.eks.amazonaws.com" cluster=remote
time="2020-12-09T13:03:00Z" level=warning msg="Gateway returned unexpected status 502. Marking as unhealthy" probe-key=crosscluster-test-2
```

```
k get links.multicluster.linkerd.io -o yaml | grep gatewayAddress
    gatewayAddress: XXXXXXXXXXXXXXXXX.us-east-2.elb.amazonaws.com
```

#### `linkerd check` output

```text
× all gateway mirrors are healthy
        liveness checks failed for crosscluster-test-2
    see https://linkerd.io/checks/#l5d-multicluster-gateways-endpoints for hints
```

### Environment

- Kubernetes Version: 1.17
- Cluster Environment: EKS
- Linkerd version: 2.9.0

**Comment by yfried** [comment]
Looks like I was missing the following value from `linkerd2-multicluster` helm chart:
```
  identityTrustDomain: cluster.local
```
This made it work


I'm super new to this, so I'm not sure if I missed something in the doc or not

**Comment by alpeb** [comment]
Thanks @yfried :+1:  This is an omission on our side so I'm reopening until it gets addressed.

**Comment by alpeb** [comment]
This actually got recently fixed by #5346 which was already merged into `main` and I'll merge into `release/2.9` to include it in the `2.9.1` release.
